share/wake/lib/system/remote_cache_api.wake

```diff
@@ -331,16 +333,53 @@ export def rscApiFindMatchingJob (req: CacheSearchRequest) (api: RemoteCacheApi)
 #  rscApiGetStringBlob (RemoteCacheBlob "asdf" "https://...") = Pass "foo\nbar\nbat" 
 # ```
 export def rscApiGetStringBlob ((CacheSearchBlob _ uri): CacheSearchBlob): Result String Error =
-    # TODO: use "small blob" for this and also parse the schema out
-    # require False = id ==* "00000000-0000-0000-0000-000000000000"
-    # else Pass  ""
+    require scheme, path, Nil = extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)` uri
+    else failWithError "rsc: uri has too many components: '{uri}'"
+
+    # maps path = "%46%6F%6F" to content = "foo"
+    def dbScheme _scheme path =
+        # If path is empty we have the empty string
+        require True = path.strlen > 0
+        else Pass ""
+
+        # If not empty then it must as lease be %00
+        require True = path.strlen > 3
+        else failWithError "rsc: Invalid db path: '{path}'"
+
+        require _empty, byteStrs = tokenize `%` path
+        else failWithError "rsc: Failed to extract bytes from db path: '{path}'"
```

Comment:
I'd either `require "", byteStrs = ...` to enforce that we don't have unsupported text in the path or prepend the potentially-not-empty string to the start of the output; probably the first since the second opens up potential injection risks.

Reply:
yeah, that's better, I just didn't think about it